nrf_security/doc/configuration.rst

```diff
@@ -7,16 +7,32 @@ Configuration
    :local:
    :depth: 2
 
-The Nordic Security Module is enabled through Kconfig with the option :kconfig:`CONFIG_NORDIC_SECURITY_BACKEND` and additional configuration is usually done with Kconfig.
+The Nordic Security Module can be enabled using `PSA driver support` or with `Legacy crypto support`.
 
-These configurations are then used to generate an Mbed TLS configuration file used during compilation.
+PSA driver support
+******************
+
+The Nordic Security Module with PSA driver support can be enabled by setting :kconfig:`CONFIG_NRF_SECURITY`, and additional configurations according to :ref:`_nrf_security_driver_config`.
+
+Enabling PSA driver support expects PSA Crypto APIs to be used.
+
+Legacy crypto support
+*********************
+
+Legacy crypto mode can be enabled by setting the Kconfig variable :kconfig:CONFIG_NORDIC_SECURITY_BACKEND`, and additional configurations according to :ref:`_nrf_security_legacy_config`.
+
+Enabling legacy crypto support allows backwards compatibility for software that require usage of Mbed TLS crypto toolbox functions prefixed with `mbedtls_`.
+
+Custom Mbed TLS configuration files
+***********************************
+
+The Kconfig options for the Nordic Security Module (nrf_security) is used to generate an Mbed TLS configuration files used during compilation.
 
 Although not recommended, it is possible to provide a custom Mbed TLS configuration file by disabling :kconfig:`CONFIG_GENERATE_MBEDTLS_CFG_FILE`.
 See :ref:`nrf_security_tls_header`.
 
```

Comment:
Are we sure CONFIG_GENERATE_MBEDTLS_CFG_FILE still works? Has anyone tested it?

Reply:
Confirmed to work, resolving